core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/twitter/TwitterConverter.java

```diff
@@ -56,28 +68,33 @@ public Tuple getNextTuple() throws TexeraException {
         }
         Tuple tuple;
         while ((tuple = inputOperator.getNextTuple()) != null) {
-            List<IField> tweetFields = generateFieldsFromJson(
-                    tuple.getField(rawDataAttribute).getValue().toString());
-            if (! tweetFields.isEmpty()) {
-                cursor++;
-                
-                List<IField> tupleFields = new ArrayList<>();
-                
-                final Tuple finalTuple = tuple;
-                tupleFields.addAll(tuple.getSchema().getAttributeNames().stream()
-                        .filter(attrName -> ! attrName.equalsIgnoreCase(rawDataAttribute))
-                        .map(attrName -> finalTuple.getField(attrName, IField.class))
-                        .collect(Collectors.toList()));
-                tupleFields.addAll(tweetFields);
-                return new Tuple(outputSchema, tupleFields);
+
+            String rawTwitterJsonString = tuple.getField(this.predicate.getTwitterJsonStringFieldName()).getValue().toString();
+            Optional<List<IField>> convertedTwitterFields = this.generateFieldsFromJson(rawTwitterJsonString);
+            if (! convertedTwitterFields.isPresent()) {
+                continue;
             }
+
```

Comment:
What is done here is correct. But I feel that the code will be much more readable if the flow is roughly like this: 

Tuple outputTuple = null;
while(input!= null) {
    String rawTwitterJsonString = .....
    Optional<List<IField>> convertedTwitterFields = this.generateFieldsFromJson(rawTwitterJsonString);
    if(convertedTwitterFields.isPresent()) {
     outputTuple = ....;
     break;}
}

return outputTuple;